internal/helper/helper.go

```diff
@@ -21,3 +21,51 @@ func GetFirstElementOf(parent string, children ...string) (key string) {
 
 	return key
 }
+
+func ReadFloat(input interface{}) float64 {
+	data, ok := input.(float64)
+	if !ok {
+		fmt.Printf("[ERROR]: Unable to covert %T to float64 \n", input)
+	}
+
+	return data
+}
+
+func ReadInt(input interface{}) int {
+	data, ok := input.(int)
+	if !ok {
+		fmt.Printf("[ERROR]: Unable to covert %T to int \n", input)
+	}
+
+	return data
+}
+
+func ReadPrimitiveTypes(input, model interface{}) {
+	switch model.(type) {
+	case *float64:
+		// Store the address of the model
+		modelAddress, _ := model.(*float64)
+		// Assign the type cast value to the model address pointed value
+		*modelAddress = input.(float64)
+	case *float32:
+		modelAddress, _ := model.(*float32)
+		*modelAddress = float32(ReadFloat(input))
+	case *int:
+		modelAddress, _ := model.(*int)
+		*modelAddress = input.(int)
+	case *int8:
+		modelAddress, _ := model.(*int8)
+		*modelAddress = int8(ReadInt(input))
+	case *int16:
+		modelAddress, _ := model.(*int16)
+		*modelAddress = int16(ReadInt(input))
+	case *int32:
+		modelAddress, _ := model.(*int32)
+		*modelAddress = int32(ReadInt(input))
+	case *int64:
+		modelAddress, _ := model.(*int64)
+		*modelAddress = int64(ReadInt(input))
+	default:
+		fmt.Printf("[ERROR]: Model type [%T] is not expected \n", model)
```

Comment:
error message needs to be more context-aware and explanatory

Reply:
Please suggest what could make it better

Reply:
Done